cirq-core/cirq/ops/conditional_operation.py

```diff
@@ -0,0 +1,154 @@
+# Copyright 2021 The Cirq Developers
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from typing import (
+    AbstractSet,
+    Any,
+    Dict,
+    FrozenSet,
+    Optional,
+    Sequence,
+    TYPE_CHECKING,
+    Tuple,
+    Union,
+)
+
+from cirq import protocols, value
+from cirq.ops import raw_types
+
+if TYPE_CHECKING:
+    import cirq
+
+
+@value.value_equality
+class ConditionalOperation(raw_types.Operation):
+    """Augments existing operations to be conditionally executed."""
+
+    def __init__(
+        self,
+        sub_operation: 'cirq.Operation',
+        conditions: Sequence[Union[str, 'cirq.MeasurementKey']],
```

Comment:
what is the meaning of multiple conditions and measurement keys that correspond to more than one value?

Reply:
I'll add some documentation here.

Multiple conditions are treated as `and`, so chaining `op.with_conditions('a', 'b').with_conditions('c', 'd')` is equivalent to `op.with_conditions('a', 'b', 'c', 'd')`.

If a measurement was of multiple qubits, all zeros is considered False, anything else is considered True.

There's a subsequent PR that will allow sympy expressions to be conditions, like `'mkey1 | mkey2'`, or `'mkey1 > 20'`. That will enable `or` conditions, and measurements-with-multiple-qubits to be used as ints,